website/static/js/profile.js

```diff
@@ -323,18 +330,25 @@ BaseViewModel.prototype.cancel = function(data, event) {
 
 BaseViewModel.prototype.submit = function() {
     if (this.hasValidProperty() && this.isValid()) {
-        $osf.putJSON(
-            this.urls.crud,
-            this.serialize()
-        ).done(
-            this.handleSuccess.bind(this)
-        ).done(
-            this.setOriginal.bind(this)
-        ).fail(
-            this.handleError.bind(this)
-        );
-    } else {
-        this.showMessages(true);
+        // Name on Name view is empty
+        if (this.nameFieldEmpty()) {
```

Comment:
As noted above, not all concrete implementations of `BaseViewModel` will have a name field. `BaseViewModel` should not have to know anything about its subclasses' fields. Maybe expose a hook (e.g. a `validateEmpty()` method that subclasses can optionally override) for validating empty fields?